### PR TITLE
fix(Action): move from actionAPI to props

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

Multiple problems:

# 1. Action is still on renderers API not the new Inject API
# 2. Action name props resolve as the actionId, so two props to do the same things.
```javascript
<Action name="my:action" />
```

is resolved but we have the following problem:

if we put actionId="my:action" and this resolve a props.name = "anAction" it will break with the error: action "anAction" not found.

`name` is an existing html props, so it should not be used to do something else.

# 3. cmf "actions" settings and API must be replace with "props" settings.

actions api has multiple problems:

* dev use them in render (it's a map state to props)
* they are less flexible than cmf props
* this api is redondant and is just used to map the settings to props.

**What is the chosen solution to this problem?**
need #1097

The CMF action api is deprecated. You should use your own component.
This api is redondant and is just used to map the settings to props.
So it will still working but this PR try to support the following migration in settings
A PR with a migration script will be done later but the idea is to go from

```javascript
{
	"actions": {
		"my-action": { ... props }
	}
}
```
to

```javascript
{
	"props": {
		"MyActionComponent#my-action": {
			...props
		}
	}
}
```

This PR ensure all Action component use the inject API

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/containers/src/actionAPI.js [hunk 1]
```diff
@@ -1,4 +1,9 @@
 import { api } from '@talend/react-cmf';
+import deprecated from '@talend/react-cmf/lib/deprecated';
+
+const msg = `@talend/react-containers/lib/actionAPI should not be used any more
```
Comment (acateland) [inline packages/containers/src/actionAPI.js]: a link to the doc would be awesome

File: packages/containers/src/Action/Action.connect.js [hunk 2]
```diff
@@ -1,41 +1,31 @@
...
-		},
-		ownProps.actionId || ownProps.name,
-	);
+	return props;
+}
 
-	props.actionId = ownProps.actionId || ownProps.name;
-	props.displayMode = info.displayMode;
+export function mergeProps(stateProps, dispatchProps, ownProps) {
+	const props = Object.assign({}, ownProps, stateProps, dispatchProps);
```
Comment (acateland) [inline packages/containers/src/Action/Action.connect.js]: beware Object.assign does a shallow merge, use https://lodash.com/docs/4.17.5#merge if it is intended to have a recursive merge

File: output/components.eslint.txt [hunk 3]
```diff
@@ -12,6 +12,13 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   1:1  error  '@storybook/react' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   4:2  error  Unexpected require()                                                                    global-require
 
+/home/travis/build/Talend/ui/packages/components/src/HeaderBar/HeaderBar.component.js
```
Comment (acateland) [inline output/components.eslint.txt]: hur

Comment (jmfrancois) [reply output/components.eslint.txt]: thanks to prettier :/

**Pull request merged**